test/Microsoft.AspNetCore.Razor.Test.Common/Language/Legacy/ClassifiedSpanVerifier.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Language.Legacy
+{
+    internal class ClassifiedSpanVerifier
+    {
+        internal static void Verify(SyntaxTreeNode node, string[] baseline)
+        {
+            using (var writer = new StringWriter())
+            {
+                var walker = new Walker(writer, baseline);
+                walker.Visit(node);
+                walker.AssertReachedEndOfBaseline();
+            }
+        }
+
+        private class Walker : ClassifiedSpanWriter
+        {
+            private readonly string[] _baseline;
+            private readonly StringWriter _writer;
+
+            private int _index;
+
+            public Walker(StringWriter writer, string[] baseline) : base(writer)
+            {
+                _writer = writer;
+                _baseline = baseline;
+            }
+
+            public override void VisitClassifiedSpan(ClassifiedSpanInternal span)
+            {
+                var expected = _index < _baseline.Length ? _baseline[_index++] : null;
+
+                _writer.GetStringBuilder().Clear();
```

Comment:
Is there a reason to use a writer here instead of just a stringbuilder?

Reply:
Yes. The writer is passed in to the base class which needs a `TextWriter`. And that base class uses it for methods like `WriteNewLine()`.